packages/graphql/src/entities/LocalProject.ts

```diff
@@ -16,6 +22,50 @@ export class LocalProject extends Project {
     return this.ctx.actions.isFirstTime(this.projectRoot, 'component')
   }
 
+  @nxs.field.nonNull.list.nonNull.type(() => Spec, {
+    args: {
+      filterBy: SpecTypeEnum,
+    },
+  })
+  async specs ({ filterBy }: NexusGenArgTypes['LocalProject']['specs']): Promise<NxsResult<'LocalProject', 'specs'>> {
+    const config = this.resolvedConfig()
+    const specs = await this.ctx.actions.getSpecs({
+      projectRoot: this.projectRoot,
+      fixturesFolder: config?.fixturesFolder?.value ?? false,
+      supportFile: config?.supportFile?.value ?? false,
+      testFiles: config?.testFiles?.value ?? [],
+      ignoreTestFiles: config?.ignoreTestFiles?.value as string[] ?? [],
+      componentFolder: config?.componentFolder?.value ?? false,
+      integrationFolder: config?.integrationFolder?.value ?? '',
+    })
+
+    debug('found specs %o, filtering by %s', specs, filterBy)
+
+    const gitinfo = await this.ctx.actions.getGitInfo(specs.map((x) => x.absolute))
```

Comment:
We need to standardize our data access patterns w/ the [DataLoader](https://github.com/graphql/dataloader) patterns - there's a lot of well established things we do for this in Services that will map well to what we're doing here - didn't want to add them all initially but I think it's time now.

For instance, if we're doing a query like:

```graphql
specs(first: 10) {
  nodes {
    id
    absolutePath
  }
}
```

And we *don't* need the git info, then we don't want to execute this. The DataLoader pattern helps a lot with this.

Reply:
Had this exact thought when I was working on this, wasn't sure if it was the time to add it, but if not now, then when? Let's do it.

Reply:
I added a fairly bad implementation of data loader, but it's working as expected. I'd like to sync on how best to access the data loaders (maybe a data context middleware like services, for example).